src/battle_util.c

```diff
@@ -6205,7 +6205,7 @@ u32 IsAbilityPreventingEscape(u32 battler)
     if (B_GHOSTS_ESCAPE >= GEN_6 && IS_BATTLER_OF_TYPE(battler, TYPE_GHOST))
         return 0;
     if ((id = IsAbilityOnOpposingSide(battler, ABILITY_SHADOW_TAG))
-        && (B_SHADOW_TAG_ESCAPE < GEN_4 && GetBattlerAbility(battler) != ABILITY_SHADOW_TAG))
+        && (B_SHADOW_TAG_ESCAPE < GEN_4 || GetBattlerAbility(battler) != ABILITY_SHADOW_TAG))
```

Comment:
Isn't the OR at the wrong place? This would make it so Shadow Tag works regardless of what `B_SHADOW_TAG_ESCAPE` is set to.

Reply:
Hmm... So the first line is saying "is there a Pokémon with Shadow Tag on the opposing side?", and then the line we're looking at should be true if that Shadow Tag should block us. So I think `(B_SHADOW_TAG_ESCAPE < GEN_4) || (GetBattlerAbility(battler) != ABILITY_SHADOW_TAG)` makes sense, it always blocks us in gen 3, or if it's not gen 3, it blocks us if we don't have Shadow Tag.

Maybe I'm wrong! This code is quite dense, perhaps it should be split up into something that's easier to understand with a quick read.

Reply:
![image](https://github.com/rh-hideout/pokeemerald-expansion/assets/35279583/cb67e006-4f91-485a-ba83-be828bf97d72)
I wrote out some truth tables in Excel to help

Reply:
> Hmm... So the first line is saying "is there a Pokémon with Shadow Tag on the opposing side?", and then the line we're looking at should be true if that Shadow Tag should block us, no? So I think `(B_SHADOW_TAG_ESCAPE < GEN_4) || (GetBattlerAbility(battler) != ABILITY_SHADOW_TAG)` makes sense, it always blocks us in gen 3, or if it's not gen 3, it blocks us if we don't have Shadow Tag.
> 
> Maybe I'm wrong! This code is quite dense, perhaps it should be split up into something that's easier to understand with a quick read.
Wouldn't Shadow Tag allow us to always switch in this case, regardless of Gen?

Reply:
> Oh! Yeah, you're right Alex—sorry! :)

np :sweat_smile:

I've tested the PR in game so it can be merged from my side. Not sure if tests are possible here.